src/components/Form.js

```diff
@@ -1,10 +1,35 @@
-const Form = () => (
-  <form method="POST" action="/" className="add">
-    <h3>ADD NEW BOOK</h3>
-    <input type="text" name="title" id="title" placeholder="Book title" />
-    <input type="text" name="author" id="author" placeholder="Author" />
-    <button type="submit">Add Book</button>
-  </form>
-);
+import { useDispatch } from 'react-redux';
+import { addBookAction } from '../redux/books/books';
+
+const Form = () => {
+  const dispatch = useDispatch();
+
+  const submitForm = (e) => {
+    e.preventDefault();
+
+    // build book obj
+    const titleInput = document.getElementById('title');
+    const authorInput = document.getElementById('author');
+    const book = {
+      title: titleInput.value,
+      author: authorInput.value,
+    };
+
+    dispatch(addBookAction(book));
+
+    // clear values
+    titleInput.value = '';
+    authorInput.value = '';
+  };
+
+  return (
+    <form method="POST" action="/" className="add" onSubmit={(e) => submitForm(e)}>
+      <h3>ADD NEW BOOK</h3>
+      <input type="text" name="title" id="title" placeholder="Book title" required />
+      <input type="text" name="author" id="author" placeholder="Author" required />
+      <button type="submit">Add Book</button>
+    </form>
+  );
+};
```

Comment:
- [ ] Kindly note that your form component should have a local state with `title`, `author`. which will be updated with the `onChange` event on your inputs.
- [ ] Accessing the dom directly is always a bad practice in REACTJS.

```jsx
// Kindly you don't need to follow this code at 100% buy at least use the same structure by updating the state(author, title) with the onChange event on your inputs.

import { useState } from 'react';
import { useDispatch } from 'react-redux';
import { addBookAction } from '../redux/books/books';

const Form = () => {
  const dispatch = useDispatch();

  const [state, setState] = useState({
    title: '',
    author: '',
  });
  const submitForm = (e) => {
    e.preventDefault();
    const { title, author } = state;
    const newBook = {
      id: Date.now(), // it's just an example feel free to generate the `id` the way you want. Just make sure the id is generated here and not in the reducer.
      title,
      author,
    };
    dispatch(addBookAction(newBook));
  };

  const handleChange = (e) => {
    const { name, value } = e.target;
    setState({ ...state, [name]: value });
  };

  return (
    <form method="POST" action="/" className="add" onSubmit={(e) => submitForm(e)}>
      <h3>ADD NEW BOOK</h3>
      <input type="text" value={state.title} name="title" id="title" placeholder="Book title" onChange={handleChange} required />
      <input type="text" value={state.author} name="author" id="author" placeholder="Author" onChange={handleChange} required />
      <button type="submit">Add Book</button>
    </form>
  );
};
```jsx